File: week_1/day-4-sliding-window/max_sum_subarray_follow_up/max_sum_subarray_stream.py
from collections import deque

stream = [2, 1, 5, 1, 3, 2]  # INPUT
k = 3

window = deque()
window_sum = 0
max_sum = float('-inf')

for num in stream:
    window.append(num)
    window_sum += num
    if len(window) > k:
        window_sum -= window.popleft()
    if len(window) == k:
        max_sum = max(max_sum, window_sum)

print(max_sum)  # Output: 9

File: week_1/day-4-sliding-window/max_sum_subarray_follow_up/maxSumSubarrayStream.js
const stream = [2, 1, 5, 1, 3, 2]; // INPUT
const k = 3;

let queue = [];
let windowSum = 0;
let maxSum = -Infinity;

for (const num of stream) {
    queue.push(num);
    windowSum += num;
    if (queue.length > k) {
        windowSum -= queue.shift();
    }
    if (queue.length === k) {
        maxSum = Math.max(maxSum, windowSum);
    }
}

console.log(maxSum); // 9
